### ISSUE TITLE
[a11y]: Content is not within a landmark element (flatpicker drop down component)

### ISSUE THREAD

**Issue by jackpronske** [carbon-design-system/carbon#15766]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.50.0

### React version

1.50.0

### Automated testing tool and ruleset

IBM Equal Accessibility Assessment Tool - Latest Version

### Description

Accessibility checks are showing that the dropdown element used within the Carbon Datepicker (flatpicker) component is not within a landmark element. After much investigation and debugging, I'm finding that this particular dropdown html element is loading OUTSIDE of the main element of the entire page. Below are two screenshots:

- Cloud Messaging UI html code, showing that the flatpicker is loading outside of the `main` element.
<img width="530" alt="Screenshot 2024-02-15 at 11 05 20 AM" src="https://github.com/carbon-design-system/carbon/assets/75291994/767ccef5-2274-4728-9125-9d0bd6137f84">

- Screenshot of https://stackblitz.com/edit/github-y8a1g6-gxjtjv?file=src%2FApp.jsx , a Stackblitz showing the isolated component also loading outside of `main` element.
<img width="521" alt="Screenshot 2024-02-15 at 12 31 11 PM" src="https://github.com/carbon-design-system/carbon/assets/75291994/9e887203-e7cd-4082-ad6f-879dcce4c423">
___
I have found that manually moving this dropdown's html element to inside the `main` content, then running the accessibility check will have it pass. I could potentially run a script that changes this everytime the page loads, but it seems that this is an unintended bug caused by the calendar component.

### Reproduction/example

https://stackblitz.com/edit/github-y8a1g6-gxjtjv?file=src%2FApp.jsx

### Steps to reproduce

1) Open react page with Carbon Calendar Component (flatpicker, range)
2) click on calendar, opening drop down
3) Open elements in dev tools, see that the dropdown html element is outside the `main` html element
3.5) run a11y check (should produce error on messaging ui page)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
@jackpronske you can use `appendTo` to attach[ it to a specific DOM node](https://stackblitz.com/edit/github-y8a1g6-yhrwwc?file=src%2Fmain.jsx). I just changed the HTML to use `main` instead of a `div` for the wrapper, and targeted the `root` ID. However, when doing that, I'm seeing a bunch of a11y violations brought up from flatpickr regarding the autogenerated `aria-labels` on each date.

**Comment by jackpronske** [comment]
This fix totally worked for temporary means! Thanks @tw15egan

**Comment by andreancardona** [comment]
@jackpronske Hey Jack I'm closing this issue - the above solution by TJ is actually how we anticipate folks to navigate Flatpicker. In the sense that it is up to the user to attach it to their specified DOM node.
